system/service/audit_service.py

```diff
@@ -92,52 +94,74 @@ def get_audit_result(self, document_uid: str) -> Optional[Dict]:
                 - auditedTime: The timestamp when the audit status was last updated.
                 - rejectedReason: The reason for rejection.
         """
-        audit = self.audit_repo.get_audit_by_document_id(document_uid)
-        if audit:
-            audit_status = self.audit_repo.get_audit_status_by_audit_status_id(audit.audit_status_id)
-            if audit_status:
-                audit_status_value = audit_status.audit_status_value
-                current_app.logger.info(f"audit_result is {audit_status_value}")
-                # possible values 1 means approved
-                if audit_status_value == 1:
-                    auditor = self.user_repo.find_user_by_username(audit_status.name)
-                    audit_result = {
-                        'auditUid': audit.uid,
-                        'documentUid': audit.document_id,
-                        'auditStatus': audit_status.audit_status_value,
-                        'auditor': auditor,
-                        'auditedTime': audit_status.updated_date.isoformat() + 'Z'
-                    }
-                # 2 means rejected
-                elif audit_status_value == 2:
-                    auditor = self.user_repo.find_user_by_username(audit_status.name)
-                    audit_result = {
-                        'auditUid': audit.uid,
-                        'documentUid': audit.document_id,
-                        'auditStatus': audit_status.audit_status_value,
-                        'rejectedReason': audit.rejected_reason,
-                        'auditor': auditor,
-                        'auditedTime': audit_status.updated_date.isoformat() + 'Z'
-                    }
-                # 3 means pending
-                elif audit_status_value == 3:
-                    auditor = self.user_repo.find_user_by_username(audit_status.name)
-                    audit_result = {
-                        'auditUid': audit.uid,
-                        'documentUid': audit.document_id,
-                        'auditStatus': audit_status.audit_status_value,
-                        'auditor': auditor
-                    }
+        document = self.document_repo.get_document_by_uid(document_uid)
+        if document:
+            document_id = document.id
+            audit = self.audit_repo.get_audit_by_document_id(document_id)
+            if audit:
+                audit_status = self.audit_repo.get_audit_status_by_audit_status_id(audit.audit_status_id)
+                if audit_status:
+                    print("audit_status", audit_status)
+                    audit_status_value = audit_status.audit_status_value
+                    print("audit_status_value", audit_status_value)
+                    current_app.logger.info(f"audit_result is {audit_status_value}")
+                    # possible values 1 means approved
+                    if audit_status_value == 1:
+                        auditor = self.user_repo.find_user_by_name(audit_status.name)
```

Comment:
should search user by `audit.creator_id`

Reply:
solved in [0785b64](https://github.com/NYCU-CloudNativeTeam/DocumentSystem/pull/11/commits/0785b64b1bbe9ef1ef4a6769914004dfdd0fdbe6)